aiosmtpd/main.py

```diff
@@ -134,23 +137,31 @@ def main(args=None):
     if args.debug > 2:
         loop.set_debug(enabled=True)
 
-    log.info('Server listening on %s:%s', args.host, args.port)
+    log.debug('Attempting to start server on %s:%s', args.host, args.port)
+    server = server_loop = None
+    try:
+        server = loop.create_server(factory, host=args.host, port=args.port)
+        server_loop = loop.run_until_complete(server)
+    except RuntimeError as e:  # pragma: nocover
+        log.critical("\u001b[33;1m" + "#" * 600 + "\u001b[0m",
+                     exc_info=e)
+        raise
+    log.debug(f"server_loop = {server_loop}")
+    log.info('Server is listening on %s:%s', args.host, args.port)
```

Comment:
This is the point where the error `Event loop stopped before Future completed.` _intermittently_ happened. The root cause was actually in `test_main.py`.

Wrapping some `log.debug` around the scene of the crime, plus a \*very\* eye-catching yellow flag ;-)